ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,13 +130,26 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        host_header = None
+        for key, value in scope["headers"]:
+            if key == b"host":
+                try:
+                    host_header = value.decode("latin-1")
+                except Exception:
+                    pass
+                break
```

Comment:
Cant you just:
```suggestion
        host_header = scope["headers"].get(b"host")
```
?

Also `latin-1` seems fishy to me. I'd think it should be `ascii`?

Reply:
Using  `get()` won't work since scope["headers"] returns a tuple not dictionary. I'll use `ascii` instead of `latin-1`

Reply:
Here's how headers are formatted: 
```
"headers": [
        (b"host", b"www.example.org"),
        (b"accept", b"application/json"), 
        .....
    ],
```
`scope["headers"].get(b"host")` raises an AttributeError

I would need to iterate through `scope["headers"]` to find `(b"host", b"www.example.org")`. My approach is a bit messy, there should be a better approach.